python/ray/ml/checkpoint.py

```diff
@@ -1,35 +1,94 @@
-import abc
-from typing import Dict, Optional
+import shutil
+import tarfile
+import tempfile
+
+import cloudpickle as pickle
+import os
+from typing import Optional
 
 import ray
+from ray.rllib.utils import DeveloperAPI
+from ray.util.ml_utils.cloud import (
+    upload_to_bucket,
+    is_cloud_target,
+    download_from_bucket,
+)
+
+
+DICT_CHECKPOINT_FILE_NAME = "dict_checkpoint.pkl"
+
+
+class Checkpoint:
+    """Ray ML Checkpoint.
 
+    This implementation provides interfaces to translate between
+    different checkpoint storage locations: Local FS storage, remote
+    node FS storage, data object, and cloud storage location.
+    """
 
-class Checkpoint(abc.ABC):
-    """Checkpoint interface"""
+    @DeveloperAPI
+    def __init__(
+        self,
+        local_path: Optional[str] = None,
+        data_dict: Optional[dict] = None,
+        uri: Optional[str] = None,
+        obj_ref: Optional[ray.ObjectRef] = None,
```

Comment:
Could you add dosctrings for the parameters?

Reply:
And it's also better to add the comments on how the parameters works togather.

Reply:
There are docstrings in the `from_xxx` methods. The constructor is private so we should avoid documenting it there